### PR TITLE
[consensus] a few steps towards RoundState

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Check commit messages for details, a few improvements.

- move pending_votes and last_vote_sent into pacemaker
- rename EventProcessor to RoundManager
- simplify a few places

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/src/epoch_manager.rs [hunk 1]
```diff
@@ -42,14 +42,13 @@ use std::{
...
-/// We suppress clippy warning here because we expect most of the time we will have EventProcessor
+/// RecoveryManager is used to process events in order to sync up with peer if we can't recover from local consensusdb
+/// RoundManager is used for normal event handling.
+/// We suppress clippy warning here because we expect most of the time we will have RoundManager
 #[allow(clippy::large_enum_variant)]
-pub enum Processor<T> {
-    SyncProcessor(SyncProcessor<T>),
-    EventProcessor(EventProcessor<T>),
+pub enum RoundProcessor<T> {
+    Recovery(RecoveryManager<T>),
```
Comment (davidiw) [inline consensus/src/epoch_manager.rs:50]: thank you

File: consensus/src/liveness/pacemaker_test.rs [hunk 2]
```diff
@@ -77,3 +99,44 @@ fn expect_timeout(round: Round, event: Option<NewRoundEvent>) {
     assert_eq!(round, event.round);
     assert_eq!(event.reason, NewRoundReason::Timeout);
 }
+
+fn generate_sync_info(
```
Comment (davidiw) [inline consensus/src/liveness/pacemaker_test.rs:103]: should we consider putting this under a #[cfg(test)] within sync_info.rs?

File: consensus/src/round_manager.rs [hunk 3]
```diff
@@ -721,17 +699,15 @@ impl<T: Payload> EventProcessor<T> {
...
-                warn!("Stop vote processing because of sync up error.");
-                return;
-            };
         }
+        if self
+            .sync_up(vote_msg.sync_info(), vote_msg.vote().author(), true)
+            .await
+            .is_err()
+        {
+            warn!("Stop vote processing because of sync up error.");
```
Comment (davidiw) [inline consensus/src/round_manager.rs:708]: interesting that we don't log the error?

File: consensus/src/epoch_manager.rs [hunk 4]
```diff
@@ -337,16 +331,16 @@ impl<T: Payload> EpochManager<T> {
             self.storage.clone(),
             self.time_service.clone(),
         );
-        processor.start().await;
-        self.processor = Some(Processor::EventProcessor(processor));
-        info!("EventProcessor started");
+        processor.start(last_vote).await;
+        self.processor = Some(RoundProcessor::Normal(processor));
+        info!("RoundManager started");
```
Comment (aching) [inline consensus/src/epoch_manager.rs:336]: Should we unify the RoundManager vs RoundProcessor naming?

Comment (zekun000) [reply consensus/src/epoch_manager.rs:336]: I have difficulty figuring out the name for the RoundProcessor, it's a enum with RecoveryManager + RoundManager, suggestions are welcome!

File: consensus/consensus-types/src/sync_info.rs [hunk 5]
```diff
@@ -41,6 +41,9 @@ impl SyncInfo {
         highest_commit_cert: QuorumCert,
         highest_timeout_cert: Option<TimeoutCertificate>,
     ) -> Self {
+        // No need to include HTC if it's lower than HQC
```
[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/consensus-types/src/sync_info.rs:44]: HQC is still useful when there's HTC, the intent here is to contain the filter logic inside the sync info constructor.

[the remark this answers is not in the document]
Comment (aching) [reply consensus/consensus-types/src/sync_info.rs:44]: I agree with the filter logic, but I'm saying we can do better to simplify things.

Currently the SyncInfo has

    highest_commit_cert: QuorumCert,
    highest_timeout_cert: Option<TimeoutCertificate>,

We could just have

enum RoundProof {
    QuorumCert(QuorumCert),
    TimeoutCert(TimeoutCertificate),    
}

At a given highest round r, if a node has both HQC_r and HTC_r, it should only pass HQC_r, it need not ever pass HTC_r.

Why does it ever need to pass both?

[the remark this answers is not in the document]
Comment (zekun000) [reply consensus/consensus-types/src/sync_info.rs:44]: even you have a HTC for round 10, a HQC for round 8 is still useful if you don't have it, can be used to propose etc.

[the remark this answers is not in the document]
Comment (aching) [reply consensus/consensus-types/src/sync_info.rs:44]: True, forgot about that.  Probably should add a comment here since it's not intuitive.

**Pull request closed without merging**